swarm.js

```diff
@@ -9,29 +10,89 @@ class Swarm extends EventEmitter {
   constructor (opts = {}) {
     super()
     const { bootstrap, ephemeral } = opts
-    this.network = guts({
+    const queue = peerQueue()
+    const network = guts({
       bootstrap,
       ephemeral,
       bind: () => {
         this.emit('listening')
       },
       socket: (socket, isTCP) => {
-        const info = peerInfo(null)
-        info.connected(socket, isTCP)
-        this.emit('connection', socket, info)
+        this._connected(socket, isTCP)
       },
       close: () => {
         this.emit('close')
       }
     })
+    const onConnect = (err, socket, isTCP) => {
+      if (err) return // todo, logging?
+      this._connected(socket, isTCP)
+    }
+    queue.on('readable', () => {
+      var peer = queue.shift()
+      while (peer) {
+        network.connect(peer, onConnect)
+        peer = queue.shift()
+      }
```

Comment:
We need to only shift from the queue until we hit a 'maxConnections' threshold.

Reply:
got it :+1: